### PR TITLE
Add _find and create document endpoints

### PR DESCRIPTION
**Pull request by fubar**
Hi Chris,
I'm testing a project that uses the _find endpoint so I added it to mock-couch (with tests). I only added its basic functionality as described in the tests.
I then realized that the endpoint for posting single documents didn't exist either, so I made a quick tweak to the _bulk_docs endpoint to support both (without explicitly testing single doc creation).
Let me know what you think.
Phil

### CODE REVIEW

File: lib/bulk_docs.js [hunk 1]
```diff
@@ -31,13 +35,7 @@ module.exports = function (self) {
       return {id: id, rev: doc._rev};
     }
 
-    if (!req.body || !req.body.docs) {
-      res.send(400, { error: 'bad_request', reason: 'invalid_json' });
-      return false;
-    }
```
Comment (fubar) [inline lib/bulk_docs.js]: Moved below

**Comment by chris-l** [comment]
Hey, thanks for your contribution, but what's `_find`?
I don't see it on the CouchDB documentation: http://docs.couchdb.org/en/1.6.1/api/database/index.html

**Comment by fubar** [comment]
Hah that is an excellent question. Turns out it is proprietary to IBM Cloudant: https://cloudant.com/blog/introducing-cloudant-query/ I'm still learning about the differences between CouchDB and Cloudant but regardless, since we're using Cloudant this is something we need. I guess I'll host this privately - unless you have a better idea.

**Comment by chris-l** [comment]
Well, certainly at this point I'm not sure about including features outside the regular couchdb. 

But, putting that aside, about what you said here:
> I then realized that the endpoint for posting single documents didn't exist either

Isn't that this: [index.js#L177](https://github.com/chris-l/mock-couch/blob/master/index.js#L77) ?

**Pull request closed without merging**